### DISCUSSION TITLE
WELCOME_SETUP=0

### DISCUSSION THREAD

**Discussion by MrVining** [GNS3/gns3-server#2202] [Ideas]
It seems like there should be:
"WELCOME_SETUP=0"
between lines 49 and 50 of remote-install.sh.

I couldn't figure out how to make a comment on the file itself.

**Comment by eantowne** [comment]
Technically, it is not required, in fact some of the other variables that are instantiated and assigned values in that portion are most likely not even necessary.

Explanation:
An if statement in a bash script that includes a non-existent variable will evaluate to "nothing", and all of the comparisons using those variables are comparing to `1`. Nothing will never == `1`, therefore not declaring the variable and declaring the variable equal to `0` in this script are functionally the same.

Could it be changed? Yes
Does it need to be changed? No
Should it be changed? Maybe?

**Reply by MrVining** [reply]
I only noticed it missing because of the error message it currently throws on a fresh install of Ubuntu 22.04 server.

**Comment by eantowne** [comment]
What error are you getting?

On Tue, Mar 21, 2023 at 21:26 MrVining ***@***.***> wrote:

> I only noticed it missing because of the error message it currently throws
> on a fresh install of Ubuntu 22.04 server.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GNS3/gns3-server/discussions/2202#discussioncomment-5387530>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZX5IYY2NBTQ3ESHJPHJRTW5JIMPANCNFSM6AAAAAAV73272Q>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Ean L. Towne

***@***.***

**Comment by MrVining** [comment]
gns3-remote-install.sh: line 304: [: ==: unary operator expected
gns3-remote-install.sh: line 457: [: ==: unary operator expected

**Comment by eantowne** [comment]
Hmmm, that is interesting, if you set the value as you suggest, does it resolve the issue?

**Reply by MrVining** [reply]
It sure does.

I should note that the install is successful regardless of the error message, or log message, whatever you want to call it.

**Comment by eantowne** [comment]
PR submitted, thank you! [PR:2211](https://github.com/GNS3/gns3-server/pull/2211)
